python/ray/worker.py

```diff
@@ -399,7 +404,16 @@ def get_gpu_ids():
         # Give all GPUs in local_mode.
         if global_worker.mode == LOCAL_MODE:
             max_gpus = global_worker.node.get_resource_spec().num_gpus
-            return global_worker.original_gpu_ids[:max_gpus]
+            assigned_ids = global_worker.original_gpu_ids[:max_gpus]
+
+    if not as_str:
+        if log_once("worker:gpu:as_str"):
+            logger.warning(
+                "ray.get_gpu_ids() will return a list of strings in a future "
+                "version of Ray. "
+                "To enable the forward-compatible behavior, use "
+                "`ray.get_gpu_ids(as_str=True)`.")
+        return [int(assigned_id) for assigned_id in assigned_ids]
```

Comment:
```suggestion
        assigned_ids = [int(assigned_id) for assigned_id in assigned_ids]
```
nit: prefer to return in a single place.

Reply:
Done